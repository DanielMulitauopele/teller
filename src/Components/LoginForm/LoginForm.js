import React, { Component } from "react";
import "./LoginForm.css";

export class LoginForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      password: ""
    };
  }

  handleSubmit = async e => {
    e.preventDefault();
    //check password against password stored in db
    //if password doesn't match, display error
    //if password matches, call logInUser
    this.props.logInUser(this.state.email);
  };

  handleChange = async e => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  render() {
    const { email, password } = this.state;
    return (
      <div className="form-wrapper">
        <form className="login-form" onSubmit={this.handleSubmit}>
          <input
            className="form-input user-email"
            name="email"
            value={email}
            onChange={this.handleChange}
            placeholder="Email"
          />
          <input
            className="form-input user-password"
            type="password"
            name="password"
            value={password}
            onChange={this.handleChange}
            placeholder="Password"
          />
          <p className="login-button" onSubmit={this.handleSubmit}>
            Go
          </p>
        </form>
      </div>
    );
  }
}

export default LoginForm;
