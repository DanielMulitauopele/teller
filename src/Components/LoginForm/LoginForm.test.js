import React from 'react'
import LoginForm from './LoginForm'
import { shallow } from 'enzyme'

describe('LoginForm Component', () => {
  let wrapper
  const logInUser = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<LoginForm logInUser={logInUser}/>)
  })

  xit('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should start with the correct default state', () => {
    expect(wrapper.state().email).toEqual('')
    expect(wrapper.state().password).toEqual('')
  })

  describe('handleSubmit function', () => {
    let mockForm

    beforeEach(() => {
      mockForm = wrapper.find('.login-form')
    })

    xit('should verify password', () => {

    })

    xit('should display error if password is incorrect', () => {

    })

    xit('should call logInUser if password is correct', () => {

    })
  })

  describe('handleChange function', () => {
    it('should set state with the correct keys and values', () => {
      const emailInput = wrapper.find('.user-email')
      const passwordInput = wrapper.find('.user-password')
      emailInput.simulate('change', {
        target: { name: 'email', value: 'someDude@gmail.com' }
      })
      passwordInput.simulate('change', {
        target: { name: 'password', value: 'yaaaaaaayyyy'}
      })

      expect(wrapper.state().email).toEqual('someDude@gmail.com')
      expect(wrapper.state().password).toEqual('yaaaaaaayyyy')
    })
  })
})