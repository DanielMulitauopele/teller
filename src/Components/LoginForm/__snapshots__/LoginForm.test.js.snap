// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginForm
    logInUser={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="login-form"
        onSubmit={[Function]}
      >
        <input
          className="form-input user-email"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          value=""
        />
        <input
          className="form-input user-password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
        <p
          className="login-button"
          onSubmit={[Function]}
        >
          Go
        </p>
      </form>,
      "className": "form-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="form-input user-email"
            name="email"
            onChange={[Function]}
            placeholder="Email"
            value=""
          />,
          <input
            className="form-input user-password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />,
          <p
            className="login-button"
            onSubmit={[Function]}
          >
            Go
          </p>,
        ],
        "className": "login-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-input user-email",
            "name": "email",
            "onChange": [Function],
            "placeholder": "Email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-input user-password",
            "name": "password",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Go",
            "className": "login-button",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": "Go",
          "type": "p",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="login-form"
          onSubmit={[Function]}
        >
          <input
            className="form-input user-email"
            name="email"
            onChange={[Function]}
            placeholder="Email"
            value=""
          />
          <input
            className="form-input user-password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
          <p
            className="login-button"
            onSubmit={[Function]}
          >
            Go
          </p>
        </form>,
        "className": "form-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="form-input user-email"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              value=""
            />,
            <input
              className="form-input user-password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />,
            <p
              className="login-button"
              onSubmit={[Function]}
            >
              Go
            </p>,
          ],
          "className": "login-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-input user-email",
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-input user-password",
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Go",
              "className": "login-button",
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": "Go",
            "type": "p",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
